### PR TITLE
Implement Float Validation

### PR DESCRIPTION
**Pull request by umithardal**
This PR implements a validation class. The implementation sweeps all dirty work under the hood and provides a clean API such that the call

`property.setValidator(DoubleValidator(float1, float2, int))`

now is sufficient for proper validation. Here, `float1`, `float2` and `int` represents the minimum, maximum and decimal precision that `property` can assume, respectively. 

The implementation now allows inputs of the form `xx.yy` for floats instead of QT way of `xx,yy`. 

The class inherits from `QDoubleValidator` directly instead of NICOS implementation (`nicos/guisupport/utils.py, line 98`). I believe this is safer in the long run. This can be changed with no effects to implementation (with probably a change in the class name), upon consensus, if necessary.

Remaning issues:

- Numerics of the form `0xyz` (with a leading `0`) are still accepted. This can be easily handled via an additional control flow, though I want to have your opinions on this.

- The use of `locale` of QT validation. At this point I can only cross my fingers and wish NICOS server will always run from the same cold cave in Sweden :)

Notes:

- A regex solution may be designed to decrease (?) the number of lines (may be) to a similar effect. However, such a solution would be bug prone as the range `[float1, float2]` is not constant (not even mentioning the horror of decimal precision).
- I was not able to refactor the code due to how `QValidate` works. I believe this is fine.
- I can remove the comments, if you find them unnecessary, though the logic was not trivial for me.
- The code is added to `utilities` and I prefer the merge it here rather than Gerrit. Because, I believe, the other residents of the module also merged here. So, we can push all of them together when the time comes. I have no strong objection to push it to Gerrit though (other than I will use this immediately for other works, so time is important).

### CODE REVIEW

File: nicos_ess/utilities/validators.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        self.setNotation(QDoubleValidator.StandardNotation)
+        self.setRange(bottom, top, decimal)  # Required for standard validation
+
+    def validate(self, string, pos):
+        # We need to add this to have a successful float casting.
+        if string == "":
+            return QValidator.Intermediate, string, pos
+        """
+        Here we hack the QT API through `Intermediate` to be able to use
```
Comment (kmurica) [inline nicos_ess/utilities/validators.py:45]: hack -> override. Hack sounds almost illegal. ;D

File: nicos_ess/utilities/validators.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        # Handle ranges in the absence of `.`, separately.
+        try:
+            if float(string) > self.maximum_value\
+                    or self.minimum_value > float(string):
+                return QValidator.Invalid, string, pos
+        except ValueError:
+            return QValidator.Invalid, string, pos
+
+        return QDoubleValidator.validate(self, string, pos)
```
Comment (kmurica) [inline nicos_ess/utilities/validators.py:76]: I tried your code with these cases and I believe they work fine.
Might be worth adding as unit tests someday:

```
from nicos.guisupport.qt import QValidator
from nicos_ess.utilities.validators import DoubleValidator

double_validator = DoubleValidator(2, 100, 4)


def test_validate_float_returns_intermediate_if_empty_string():
    stat, _, _ = double_validator.validate("", 0)
    assert stat == QValidator.Intermediate


def test_validate_float_returns_invalid_if_string_not_float():
    stat, _, _ = double_validator.validate("12.34.55.2", 0)
    assert stat == QValidator.Invalid


def test_validate_float_returns_invalid_if_value_lower_than_min_val():
    stat, _, _ = double_validator.validate("1.5", 0)
    assert stat == QValidator.Invalid


def test_validate_integer_returns_invalid_if_value_lower_than_min_val():
    stat, _, _ = double_validator.validate("0", 0)
    assert stat == QValidator.Invalid


def test_validate_float_returns_invalid_if_value_higher_than_max_val():
    stat, _, _ = double_validator.validate("1.5", 0)
    assert stat == QValidator.Invalid


def test_validate_integer_returns_invalid_if_value_higher_than_max_val():
    stat, _, _ = double_validator.validate("0", 0)
    assert stat == QValidator.Invalid


def test_validate_alphabetic_input_returns_invalid():
    stat, _, _ = double_validator.validate("invalid", 0)
    assert stat == QValidator.Invalid


def test_validate_valid_integer_returns_acceptable():
    stat, _, _ = double_validator.validate("10", 0)
    assert stat == QValidator.Acceptable
```

File: nicos_ess/utilities/validators.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        leads acceptable values like
+        
+            >> 12.sdaad12r..asf-dsfs1*,
+            
+        which are of course invalid. 
+        """
+        if '.' in string:
+            try:
+                if self.maximum_value > float(string) > self.minimum_value:
+                    return QValidator.Intermediate, string, pos
```
Comment (kmurica) [inline nicos_ess/utilities/validators.py:61]: Question: why is this not returning acceptable? Also should it not check for decimal precision?

Comment (umithardal) [reply nicos_ess/utilities/validators.py:61]: Thanks. Concerning `Intermadiate -> Acceptable` both works as `Intermediate` for QT is also `Acceptable` at that specific point (but not as a final) :) Yeah weird huh?  There are of course some differences under the hood concerning call stack. However, decimal precision fails without the `Finally` block, even with your suggestion of `Acceptable`.  I am not sure why, but figuring out putting a `Finally` block took me some time. :)

My logic was keep it as `let us assume this float is acceptable` then it is definitely so. I will change it to `Acceptable` though, seems better.

File: nicos_ess/utilities/validators.py (hunk 1 above)
Comment (umithardal) [reply nicos_ess/utilities/validators.py:45]: :) I will find a better description or may be remove this part of the comment as I am changing it to `Acceptable`.

File: nicos_ess/utilities/validators.py (hunk 2 above)
Comment (umithardal) [reply nicos_ess/utilities/validators.py:76]: However I need this class as soon as possible to finish up, so I would appreciate a separate PR for the tests.

**Review by kmurica — APPROVED** [review]
LGTM

**Pull request merged**